### PR TITLE
Auto loader for per-commit tests

### PR DESCRIPTION
**Pull request by faizkhan00**

### CODE REVIEW

File: ethereumj-core/src/main/java/org/ethereum/jsontestsuite/JSONReader.java [hunk 1]
```diff
@@ -71,6 +73,36 @@ public static String getFromUrl(String urlToRead) {
...
+            for (Object oEntry : tree) {
+                JSONObject entry = (JSONObject) oEntry;
+                String testName = (String) entry.get("path");
+                if ( testName.equals(testcase) ) {
+                    String blobresult = getFromUrl( (String) entry.get("url") );
+
+                    testSuiteObj = (JSONObject) parser.parse(blobresult);
+                    String blob  = (String) testSuiteObj.get("content");
+                    byte[] valueDecoded= Base64.decodeBase64(blob.getBytes() );
+                    //System.out.println("Decoded value is " + new String(valueDecoded));
```
Comment (romanman) [inline ethereumj-core/src/main/java/org/ethereum/jsontestsuite/JSONReader.java]: Curious why the 64 encode it ?

Comment (faizkhan00) [inline ethereumj-core/src/main/java/org/ethereum/jsontestsuite/JSONReader.java]: If you check the file, https://api.github.com/repos/ethereum/tests/git/blobs/e9894c621f1703df6066efecdf318b32e1c59fea

you can see they are all encoded base64, and I haven't seen any other encoding formats from the repo, but
its possible to add those if they exist too
```
{
  "sha": "e9894c621f1703df6066efecdf318b32e1c59fea",
  "size": 1517,
  "url": "https://api.github.com/repos/ethereum/tests/git/blobs/e9894c621f1703df6066efecdf318b32e1c59fea",
  "content": "ewogICAgInJhbmRvbVZNdGVzdCIgOiB7CiAgICAgICAgImNhbGxjcmVhdGVz\nIiA6IFsKICAgICAgICBdLAogICAgICAgICJlbnYiIDogewogICAgICAgICAg\nICAiY3VycmVudENvaW5iYXNlIiA6ICIyYWRjMjU2NjUwMThhYTFmZTBlNmJj\nNjY2ZGFjOGZjMjY5N2ZmOWJhIiwKICAgICAgICAgICAgImN1cnJlbnREaWZm\naWN1bHR5IiA6ICIyNTYiLAogICAgICAgICAgICAiY3VycmVudEdhc0xpbWl0\nIiA6ICIxMDAwMDAwIiwKICAgICAgICAgICAgImN1cnJlbnROdW1iZXIiIDog\nIjAiLAogICAgICAgICAgICAiY3VycmVudFRpbWVzdGFtcCIgOiAiMSIsCiAg\nICAgICAgICAgICJwcmV2aW91c0hhc2giIDogIjVlMjBhMDQ1M2NlY2QwNjVl\nYTU5YzM3YWM2M2UwNzllZTA4OTk4YjYwNDUxMzZhOGNlNjYzNWM3OTEyZWMw\nYjYiCiAgICAgICAgfSwKICAgICAgICAiZXhlYyIgOiB7CiAgICAgICAgICAg\nICJhZGRyZXNzIiA6ICIwZjU3MmU1Mjk1YzU3ZjE1ODg2ZjliMjYzZTJmNmQy\nZDZjN2I1ZWM2IiwKICAgICAgICAgICAgImNhbGxlciIgOiAiY2QxNzIyZjM5\nNDdkZWY0Y2YxNDQ2NzlkYTM5YzRjMzJiZGMzNTY4MSIsCiAgICAgICAgICAg\nICJjb2RlIiA6ICIweDY3ZjE4MjE2NTQ0MDliNzMwMzY0ODFmMzY2MTMiLAog\nICAgICAgICAgICAiZGF0YSIgOiAiMHgiLAogICAgICAgICAgICAiZ2FzIiA6\nICIxMDAwMCIsCiAgICAgICAgICAgICJnYXNQcmljZSIgOiAiMTAwMDAwMDAw\nMDAwMDAwIiwKICAgICAgICAgICAgIm9yaWdpbiIgOiAiY2QxNzIyZjM5NDdk\nZWY0Y2YxNDQ2NzlkYTM5YzRjMzJiZGMzNTY4MSIsCiAgICAgICAgICAgICJ2\nYWx1ZSIgOiAiMTAwMDAwMDAwMDAwMDAwMDAwMCIKICAgICAgICB9LAogICAg\nICAgICJnYXMiIDogIjk5OTgiLAogICAgICAgICJsb2dzIiA6IFsKICAgICAg\nICBdLAogICAgICAgICJvdXQiIDogIjB4IiwKICAgICAgICAicG9zdCIgOiB7\nCiAgICAgICAgICAgICIwZjU3MmU1Mjk1YzU3ZjE1ODg2ZjliMjYzZTJmNmQy\nZDZjN2I1ZWM2IiA6IHsKICAgICAgICAgICAgICAgICJiYWxhbmNlIiA6ICIx\nMDAwMDAwMDAwMDAwMDAwMDAwIiwKICAgICAgICAgICAgICAgICJjb2RlIiA6\nICIweDY3ZjE4MjE2NTQ0MDliNzMwMzY0ODFmMzY2MTMiLAogICAgICAgICAg\nICAgICAgIm5vbmNlIiA6ICIwIiwKICAgICAgICAgICAgICAgICJzdG9yYWdl\nIiA6IHsKICAgICAgICAgICAgICAgIH0KICAgICAgICAgICAgfQogICAgICAg\nIH0sCiAgICAgICAgInByZSIgOiB7CiAgICAgICAgICAgICIwZjU3MmU1Mjk1\nYzU3ZjE1ODg2ZjliMjYzZTJmNmQyZDZjN2I1ZWM2IiA6IHsKICAgICAgICAg\nICAgICAgICJiYWxhbmNlIiA6ICIxMDAwMDAwMDAwMDAwMDAwMDAwIiwKICAg\nICAgICAgICAgICAgICJjb2RlIiA6ICIweDY3ZjE4MjE2NTQ0MDliNzMwMzY0\nODFmMzY2MTMiLAogICAgICAgICAgICAgICAgIm5vbmNlIiA6ICIwIiwKICAg\nICAgICAgICAgICAgICJzdG9yYWdlIiA6IHsKICAgICAgICAgICAgICAgIH0K\nICAgICAgICAgICAgfQogICAgICAgIH0KICAgIH0KfQo=\n",
  "encoding": "base64"
}
```

Comment (romanman) [inline ethereumj-core/src/main/java/org/ethereum/jsontestsuite/JSONReader.java]: So those are files/directories that are part of the commit ?

Comment (faizkhan00) [inline ethereumj-core/src/main/java/org/ethereum/jsontestsuite/JSONReader.java]: Yep, they are the individual test cases for the specific commit

Comment (romanman) [inline ethereumj-core/src/main/java/org/ethereum/jsontestsuite/JSONReader.java]: question: is it include also the files that are no changed in that
particular commit ?

Comment (faizkhan00) [inline ethereumj-core/src/main/java/org/ethereum/jsontestsuite/JSONReader.java]: Yeah, unchanged files will still be a part of the commit

Comment (romanman) [inline ethereumj-core/src/main/java/org/ethereum/jsontestsuite/JSONReader.java]: cool thanks , good job.

**Pull request merged**